python/pyiceberg/catalog/__init__.py

```diff
@@ -600,3 +600,7 @@ def _get_updated_props_and_update_summary(
         )
 
         return properties_update_summary, updated_properties
+
+    def __repr__(self) -> str:
+        """Returns the string representation of the Catalog class."""
+        return f"{self.__class__} with name: {self.name}"
```

Comment:
What do you think of?
```suggestion
        return f"{self.name} ({self.__class__})"
```

Reply:
Definitely cleaner!